### PR TITLE
CBMC: Update to Version 6

### PR DESCRIPTION
**Pull request by hanno-becker**
Bump CBMC to [freshly released](https://github.com/diffblue/cbmc/releases/tag/cbmc-6.0.0) version 6.

### CODE REVIEW

**Comment by hanno-becker** [comment]
@potsrevennil @mkannwischer Out of curiosity I added the `benchmark` tag here, but it didn't trigger the CI. Is this deliberate?

**Comment by potsrevennil** [comment]
This is wierd, this should not happened. I expected that after adding the benchmark label, click rerun job, then the bench workflow should just run. I'll investigate this tmr.

**Comment by mkannwischer** [comment]
> This is wierd, this should not happened. I expected that after adding the benchmark label, click rerun job, then the bench workflow should just run. I'll investigate this tmr.

Here is another think to look at: https://github.com/pq-code-package/mlkem-c-aarch64/actions/runs/9595340954 reports: 

```
MLKEM512
keypair cycles=139579
encaps cycles=173250
decaps cycles=223791
CRYPTO_SECRETKEYBYTES: 1632
CRYPTO_PUBLICKEYBYTES: 800
CRYPTO_CIPHERTEXTBYTES: 768

MLKEM768
keypair cycles=0
encaps cycles=0
decaps cycles=0
CRYPTO_SECRETKEYBYTES: 2400
CRYPTO_PUBLICKEYBYTES: 1184
CRYPTO_CIPHERTEXTBYTES: 1088

MLKEM1024
keypair cycles=377669
encaps cycles=428755
decaps cycles=512516
CRYPTO_SECRETKEYBYTES: 3168
CRYPTO_PUBLICKEYBYTES: 1568
CRYPTO_CIPHERTEXTBYTES: 1568
```

**Comment by hanno-becker** [comment]
@cothan Let's avoid merge commits if possible, and instead rebase PRs.

**Comment by cothan** [comment]
@hanno-becker , got it. I wish there is a rebase button at the Github UI.

File: cbmc/proofs/Makefile.common [hunk 1]
```diff
@@ -517,7 +517,7 @@ COMMA :=,
 CBMCFLAGS += --object-bits $(CBMC_OBJECT_BITS)
 
 # --object_bits is removed in goto-cc 6.0.0
-COMPILE_FLAGS += --object-bits $(CBMC_OBJECT_BITS)
+# COMPILE_FLAGS += --object-bits $(CBMC_OBJECT_BITS)
```
Comment (cothan) [inline cbmc/proofs/Makefile.common:520]: Remove the comment

**Review by cothan — APPROVED** [review]

**Comment by rod-chapman** [comment]
I have reproduced all proofs here from this branch, and all is well, so happy.

**Comment by cothan** [comment]
@hanno-becker , I change the github setting, there is a button to rebase the branch in Github UI. 

Screenshot 2024-06-20 at 11 46 22 AM

**Comment by potsrevennil** [comment]
@hanno-becker I have a question maybe not that related to this PR. 

I noticed that in here
https://github.com/pq-code-package/mlkem-c-aarch64/blob/95615fc0c742cf4b3ace32e0578d830d9447751a/.github/workflows/ci.yml#L117-L122

Since we are using `bash` with the `-e` flag, seems like if the previous proof failed, the following proof will not be executed, is this as expected ?

**Comment by hanno-becker** [comment]
@potsrevennil I don't think this was intentional, but upon consideration, it may be the right call -- those CBMC proofs tend to be computationally expensive, and each line still conducts all proofs for a fixed K=2,3,4.

File: cbmc/proofs/Makefile.common (hunk 1 above)
Comment (hanno-becker) [inline cbmc/proofs/Makefile.common:520]:
```suggestion
```

**Comment by potsrevennil** [comment]
@hanno-becker OK, make sense! Other parts LGTM. 
There is still some issue for the benchmarking where @mkannwischer pointed out, but it can be further discussed in other issue or pr I think.

**Review by potsrevennil — APPROVED** [review]

**Comment by hanno-becker** [comment]
@potsrevennil Yes, I agree we don't want merge commits (see comments above). I rebased now.

**Review by potsrevennil — APPROVED** [review]

**Comment by potsrevennil** [comment]
@hanno-becker Thanks!

**Pull request merged**